nncf/compression_method_api.py

```diff
@@ -51,14 +51,20 @@ class PTCompressionLoss(nn.Module, CompressionLoss):
     sparsity algorithm calculates the number of non-zero weights in convolutional
     and fully-connected layers to construct the loss function.
     """
+    def __init__(self, default_device=None):
+        """
+        :param default_device: torch device being used for creating a default value of loss on the target device
```

Comment:
I am fairly sure that everywhere we have to fall back on the default value of the loss we also know the device of the model to be compressed. I suggest that instead of changing the API for this you keep the API as-is and instead define a new `ZeroCompressionLoss(PTCompressionLoss)` with an init function taking a device. Instantiate it explicitly in all algos that do not define their specific compression loss.

Reply:
With current design I see the problem with implementation of PTCompressionAlgorithmController:
```
def __init__(self, target_model: ModelType):
     self._loss = PTCompressionLoss()
```
Algorithms are not required to define a new ZeroCompressionLoss. That still would lead to errors with device.

Reply:
Even the `PTCompressionAlgorithmController` should be an interface, not a real base class. The real controller class specifying no compression is `NoCompressionAlgorithmController`.

Reply:
it means that we need to change current design first (probably in a separate PR), because this class has a lot of implemented methods: `export_model`, `run_batchnorm_adaptation`, `statistics`

Reply:
Will be done in a separate PR.